src/mlpack/tests/split_data_test.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file sparse_autoencoder_test.cpp
+ * @author Siddharth Agrawal
+ *
+ * Test the SparseAutoencoder class.
+ */
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/split_data.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "old_boost_test_definitions.hpp"
+
+using namespace mlpack;
+using namespace arma;
+
+BOOST_AUTO_TEST_SUITE(SplitDataTest);
+
+void compareData(arma::mat const &inputData, arma::mat const &compareData,
+                 arma::Row<size_t> const &inputLabel)
+{
+    for(size_t i = 0; i != compareData.n_cols; ++i){
+        arma::mat const &lhsCol = inputData.col(inputLabel(i));
+        arma::mat const &rhsCol = compareData.col(i);
+        for(size_t j = 0; j != lhsCol.n_rows; ++j){
+            BOOST_REQUIRE_CLOSE(lhsCol(j), rhsCol(j), 1e-5);
+        }
+    }
+}
```

Comment:
Do you mind adding a comment saying what this function does?  It might also be worth noting that this function depends on the labels being equivalent to the original point index (the trick you use in line 36/37).